components/browser/icons/src/main/java/mozilla/components/browser/icons/decoder/AndroidIconDecoder.kt

```diff
@@ -35,19 +36,27 @@ class AndroidIconDecoder : IconDecoder {
 
                 bitmap.width * (1.0f / maxScaleFactor) > maxSize ||
                     bitmap.height * (1.0f / maxScaleFactor) > maxSize -> {
-                    logger.debug("BitmapFactory returned way too large image")
-                    null
+                    val scale = maxSize.toFloat() / max(bitmap.width, bitmap.height)
+                    createScaledBitmap(
+                        bitmap,
+                        (bitmap.width * scale).toInt(),
+                        (bitmap.height * scale).toInt()
+                    )
```

Comment:
Please let's not scale here. This is planned to be added as an `IconProcessor` in #2041.